src/_bentoml_sdk/_pydantic.py

```diff
@@ -53,13 +53,14 @@ def numpy_prepare_pydantic_annotations(
     _, remaining_annotations = _known_annotated_metadata.collect_known_metadata(
         annotations
     )
-    for i, annotation in enumerate(remaining_annotations[:]):
+    orig_len = len(remaining_annotations)
+    for i, annotation in enumerate(reversed(remaining_annotations[:])):
```

Comment:
```suggestion
    for i, annotation in reversed(enumerate(remaining_annotations[:])):
```

So you don't need to calculate the actual index

Reply:
@frostming Does that work for you? I get a `TypeError` with Python 3.9:

```python
>>> my_list = ["a", "b"]
>>> for i, e in reversed(enumerate(my_list[:])):
...     print(i, e)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'enumerate' object is not reversible
```

Reply:
It looks like this works:

```python
>>> my_list = ["a", "b"]
>>> for i, e in reversed(list(enumerate(my_list[:]))):
...     print(i, e)
... 
1 b
0 a
```

Reply:
Fine, but `[:]` is no necessary then, can you remove it?

Reply:
Yes, good catch. Thanks for the feedback. I've removed it.